### PR TITLE
Check QGIS Server URL and get QGIS Server version information

### PR DESCRIPTION
**Pull request by rldhont**
With a simple QGIS project, Lizmap can test QGIS Server URL and get release name and version number.

### CODE REVIEW

**Comment by rldhont** [comment]
@Gustry We can get QGIS Server version information with simple virtual field and a WFS GetFeature Request.

**Comment by rldhont** [comment]
@laurentj any suggestion about the place to put the QGIS Project to get QGIS Server version information.

**Comment by Gustry** [comment]
Question, it's already in the Lizmap server plugin
SERVICE=LIZMAP
REQUEST=GetServerSettings
```json
{
  "qgis": {
    "version": "3.13.0",
    "name": "Master"
  },
  "services": [
    "WMS",
    "WFS",
    "WCS",
    "WMTS",
    "ATLAS",
    "EXPRESSION",
    "LIZMAP"
  ],
  "lizmap": {
    "name": "Lizmap",
    "version": "3.3.99-beta"
  }
}
```
So it's jsut in the meantime?

Small documentation https://github.com/3liz/lizmap-plugin/#lizmap-server-api I made it quickly.

**Comment by Gustry** [comment]
Nice idea!

**Comment by rldhont** [comment]
@Gustry it is a way to get it without plugin.

**Comment by laurentj** [comment]
@rldhont I'm not sure I understand all the code about this "fake" project. But if I well understood, you assume that Qgis server has an access to this qgis_info.qgs file, right?

If this is a case, I think it won't work in many case :

- when the `rootRepository` is set in the configuration
- when the QGIS server is installed in an other server/container, and only the directory  of real qgis projects is shared between servers/containers.

Anyway, I don't like this way to get informations about qgis server. I would prefer to enhance Qgis server or to create a plugin for Qgis Server. This could be the first feature to implement for the Qgis plugin we talked about weeks ago.

File: lizmap/modules/admin/controllers/config.classic.php [hunk 1]
```diff
@@ -205,6 +210,17 @@ public function saveServices()
             $ok = false;
         }
 
+        // Check the WMS Server URL
+        $wmsServerURL = $form->getData('wmsServerURL');
+        jClasses::inc('lizmap~qgisServer');
```
Comment (laurentj) [inline lizmap/modules/admin/controllers/config.classic.php:215]: we should declare qgisServer as an autoloaded class into module.xml, and avoid to use jClasses.

**Comment by rldhont** [comment]
> @rldhont I'm not sure I understand all the code about this "fake" project. But if I well understood, you assume that Qgis server has an access to this qgis_info.qgs file, right?
> 
> If this is a case, I think it won't work in many case :
> 
>     * when the `rootRepository` is set in the configuration
> 
>     * when the QGIS server is installed in an other server/container, and only the directory  of real qgis projects is shared between servers/containers.
> 
> 
> Anyway, I don't like this way to get informations about qgis server. I would prefer to enhance Qgis server or to create a plugin for Qgis Server. This could be the first feature to implement the Qgis plugin we talked about weeks ago.

The `rootRepository` is not a problem, because the project is not a lizmap one just a QGIS one.

The idea is to get QGIS Server information without plugins. The plugin code has been already done [1](https://github.com/3liz/lizmap-plugin/pull/196) but all Lizmap users do not install plugins and we have some difficulties to know QGIS Server version.

It is not a fake project but it has to be accessible for QGIS Server.

File: lizmap/modules/admin/controllers/config.classic.php (hunk 1 above)
Comment (rldhont) [reply lizmap/modules/admin/controllers/config.classic.php:215]: `qgisServer` is also used with `jClasses::getService` https://github.com/3liz/lizmap-web-client/blob/master/lizmap/modules/lizmap/classes/lizmapProject.class.php#L684. can `qgisServer` be an autoloaded classes ?

**Comment by laurentj** [comment]
>It is not a fake project but it has to be accessible for QGIS Server.

So it won't work at all on our servers

**Comment by mdouchin** [comment]
We could probably easily use this as a alternate solution when no information could have been fetched from the server lizmap plugin. Check QGIS Server version in this order

* attempt to get information from lizmap plugin (server side).
* if lizmap plugin not installed, try to fetch it from this QGIS project
* if this QGIS project is not accessible, trust the configuration `lizmapConfig.ini.php`

**Comment by rldhont** [comment]
> > It is not a fake project but it has to be accessible for QGIS Server.
> 
> So it won't work at all on our servers

Do you have an idea to provide this project ?

It will be necessary to have a QGIS project to get QGIS Server version information by a plugin and to get plugins informations.

**Comment by mdouchin** [comment]
What about puttin this project in the QGIS Server VM/container/etc., with the environment `QGIS_PROJECT_FILE` variable to set this project path. If I understand correctly, QGIS Server will first look for some environment variable to get the project, which is overriden when using the MAP parameter ? If the QGIS project lies in the same server as where QGIS Server is installed, no issue then ?
See https://docs.qgis.org/3.10/en/docs/user_manual/working_with_ogc/server/config.html#qgis-project-file

**Comment by rldhont** [comment]
@mdouchin With or without `QGIS_PROJECT_FILE`, it will be great to have a way to install this QGIS project and its data somewhere QGIS Server can access it.

@laurentj what is your recommendation about installing the QGIS project somewhere QGIS Server can access it but outside the FTP directory ?

**Comment by nboisteault** [comment]
@rldhont  @laurentj @mdouchin Do I tag this PR to 3.5 ?

**Comment by rldhont** [comment]
Done @nboisteault

**Comment by Gustry** [comment]
Can we revive this pull request and find a solution?
It's very needed to have this information somehow.

I wish we can have a warning message bar for admins saying that they might have issues. 🙏

QGIS 3.16 changed the way to save the "Do not expose in WMS/WFS". QGIS Server < 3.16 is absolutely not able to read this configuration.

**Comment by nboisteault** [comment]
@rldhont should we tag this PR for release 3.6?

**Comment by Gustry** [comment]
IMHO, this PR is a dead end since https://github.com/3liz/lizmap-plugin/pull/335

Should we close it ?

**Pull request closed without merging**